src/arrayMethodJoin.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let res = '';
+    const sep = separator !== undefined ? separator : ',';
+    const arr = [...this];
+
+    for (let i = 0; i < arr.length; i++) {
+      if (
+        (arr[i] || arr[i] === 0 || isNaN(arr[i]) || arr[i] === false)
+        && arr[i] !== undefined
+      ) {
+        res += String(arr[i]);
+      }
+      res += sep;
```

Comment:
One blank line between a block and the next command
```suggestion
      }
      
      res += sep;
```